test/e2e/tests/confirmations/signatures/malicious-signatures.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { strict as assert } from 'assert';
+import { Suite } from 'mocha';
+import { MockedEndpoint } from 'mockttp';
+import { WINDOW_TITLES, openDapp, unlockWallet } from '../../../helpers';
+import { Driver } from '../../../webdriver/driver';
+import {
+  scrollAndConfirmAndAssertConfirm,
+  withRedesignConfirmationFixtures,
+} from '../helpers';
+import { TestSuiteArguments } from '../transactions/shared';
+import { assertSignatureRejectedMetrics } from './signature-helpers';
+
+describe('Malicious Confirmation Signature - Bad Domain @no-mmi', function (this: Suite) {
+  it('displays alert for domain binding and confirms', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver }: TestSuiteArguments) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await driver.clickElement('#siweBadDomain');
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        const alert = await driver.findElement('[data-testid="inline-alert"]');
```

Comment:
Pending the full page object model, could large parts of this go into reusable helpers such as `expectAlert` or `acknowledgeAlert` or `confirmWithAlertModal`?

Reply:
done